gitlab/mixins.py

```diff
@@ -231,7 +231,17 @@ def list(self, **kwargs: Any) -> Union[base.RESTObjectList, List[base.RESTObject
             for attr_name, type_cls in self._types.items():
                 if attr_name in data.keys():
                     type_obj = type_cls(data[attr_name])
-                    data[attr_name] = type_obj.get_for_api()
+                    if isinstance(type_obj, g_types.ArrayAttribute):
```

Comment:
I wonder if this checking should be done in some helper function where we pass in a dictionary and it returns back a List[Tuple[str, str]] data structure, which can then go to `requests.request`

I'm almost thinking we should defer all of this until we get to `gitlab/client.py:http_request()`. But the problem is that by that point we don't have the `self._types` structure 😟

Reply:
Done